### PR TITLE
Allow setting of http timeout in the session

### LINKED ISSUE #10: be able to set http timeout
currently the session is using default http client, which means no timeout.

### PR DESCRIPTION
**Pull request by renier**
Fixes #10

### CODE REVIEW

File: session/xmlrpc.go [hunk 1]
```diff
@@ -55,13 +55,10 @@ func (mrt debugRoundTripper) RoundTrip(request *http.Request) (*http.Response, e
 }
 
 // XML-RPC Transport
-type XmlRpcTransport struct {
-	Timeout time.Duration
-}
+type XmlRpcTransport struct{}
 
-const DefaultXmlRpcTimeout = time.Second * 30
+const DefaultXmlRpcTimeout = time.Second * 60
```
Comment (riethm) [inline session/xmlrpc.go]: Are we intentionally allowing transport handlers to have different default timeouts?

File: session/xmlrpc.go [hunk 2]
```diff
@@ -82,8 +79,8 @@ func (x *XmlRpcTransport) DoRequest(
 		}
 
 		timeout := DefaultXmlRpcTimeout
-		if x.Timeout != 0 {
-			timeout = x.Timeout
+		if sess.Timeout != 0 {
```
Comment (riethm) [inline session/xmlrpc.go]: Not sure I like that it is essentially impossible to override timeout to be 0 (unlimited).

File: session/xmlrpc.go (hunk 1 above)
Comment (renier) [inline session/xmlrpc.go]: Not intentionally. You want to have one for all transports?

File: session/xmlrpc.go (hunk 2 above)
Comment (renier) [inline session/xmlrpc.go]: timeout should never be unlimited. it is a potential problem for both client and server. users can come up with a really long timeout if they wish.

File: session/xmlrpc.go (hunk 1 above)
Comment (riethm) [inline session/xmlrpc.go]: yeah, at least for the library-provided transports, consistency is nice.

Comment (renier) [inline session/xmlrpc.go]: 8be6627

**Pull request merged**